File: pe/nonvisualstudio/clojure/Spec/Problem0005.Spec.clj
;(load-file "C:/Users/Thomas/Documents/GitHub/practice/pe/nonvisualstudio/clojure/Spec/Problem0005.Spec.clj")

(ns pe.clojure.implementation.problem0005.spec)

(load-file "C:/Users/Thomas/Documents/GitHub/practice/pe/nonvisualstudio/clojure/Implementation/Problem0005.clj")
(use 'pe.clojure.implementation.problem0005)
(use 'clojure.test)

(deftest Problem0005

  (testing "lcmOfList"
    (is (= 2520 (lcmOfList 10)))
    (is (= 27720 (lcmOfList 11)))
    (is (= 232792560 (lcmOfList 20)))))


(run-tests)

File: pe/nonvisualstudio/elixir/Spec/Problem0005.Spec.ex
# IN POWERSHELL
# iex.bat - REPL
# elixir c:\Users\Thomas\Documents\GitHub\practice\pe\nonvisualstudio\elixir\Spec\Problem0005.Spec.ex

Code.require_file "../Implementation/Problem0005.ex", __DIR__

ExUnit.start

defmodule Problem0005Tests do
  use ExUnit.Case, async: true

  test "lcmOfList(10) should return 2520." do
    assert Problem0005.lcmOfList(10) == 2520
  end

  test "lcmOfList(11) should return 27720." do
    assert Problem0005.lcmOfList(11) == 27720
  end

  test "lcmOfList(20) should return 232_792_560." do
    assert Problem0005.lcmOfList(20) == 232_792_560
  end
end
